src/test/java/kz/insar/checkbinance/containers/BinanceAPIHelper.java

```diff
@@ -165,29 +148,96 @@ private BinanceAPIHelper mockRequestLegacyLastPrice(
                 response);
     }
 
-    private BinanceAPIHelper mockRequestLegacyLastPrice(
+    public BinanceAPIHelper mockRequestLegacyLastPrice(
             String requestSymbol, int requestLimit, BNBLegacyLastPriceResponse response) {
         return mockRequestLegacyLastPrice(requestSymbol, requestLimit,
-                response().withBody(JsonBody.json(response.toRecentTradeDTOs)).withStatusCode(200));
+                response().withBody(JsonBody.json(response.toRecentTradeDTOs())).withStatusCode(200));
+    }
+
+    public BinanceAPIHelper mockRequestLegacyLastPrice(String requestSymbol, BNBLegacyLastPriceResponse response) {
+        return mockRequestLegacyLastPrice(requestSymbol, response.getPricesQuantity(), response);
+    }
+
+    public BinanceAPIHelper mockRequestLegacyLastPrice(
+            List<String> requestSymbols, int requestLimit, List<BNBLegacyLastPriceResponse> responses) {
+        if (requestSymbols.size() != responses.size()) throw new IllegalArgumentException("request symbols and responses doesn't match");
+        for (int i = 0; i < responses.size(); i++) {
+            mockRequestLegacyLastPrice(requestSymbols.get(i), requestLimit, responses.get(i));
+        }
+        return this;
+    }
+
+    public BinanceAPIHelper mockRequestLegacyLastPrice(List<String> requestSymbols, List<BNBLegacyLastPriceResponse> responses) {
+        var limit = responses.get(0).getPricesQuantity();
+        if (!responses.stream().allMatch(response -> Objects.equals(response.getPricesQuantity(), limit)))
+            throw new IllegalArgumentException("Different responses' prices size"); //todo some uncertainty exception
+        return mockRequestLegacyLastPrice(requestSymbols, limit, responses);
+    }
+
+    public BinanceAPIHelper mockRequestLegacyLastPrice(List<BNBLegacyLastPriceResponse> responses) {
+        return mockRequestLegacyLastPrice(responses.stream().map(BNBLegacyLastPriceResponse::getCommonSymbolName).collect(Collectors.toList()), responses);
```

Comment:
getCommonSymbolName should be plural

Reply:
getCommonSymbolName returns only one symbol name because BinanceAPI requests only one symbol and limit. BNBLegacyLastPriceResponse could contain trades with different symbol names to test bad cases, so this method tries to find common name beyond trades. On fail it throws exception to specify request symbol